### PR TITLE
Allow number format for slider

### PR DESCRIPTION
**Pull request by ethanroday**
Addresses #7526.

Introduces a new optional prop on the `Slider` component called `valueFormat`. It is a function that can be used to format the value label.

### CODE REVIEW

File: common/changes/office-ui-fabric-react/slider-number-format_2019-01-04-19-23.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+  "changes": [
+    {
+      "packageName": "office-ui-fabric-react",
+      "comment": "Allows a custom value format function to be passed to the Slider component",
```
Comment (Vitalius1) [inline common/changes/office-ui-fabric-react/slider-number-format_2019-01-04-19-23.json]: nit: could you please change the comment to start with _'Slider: .....'_? Makes it easier to scan through the release notes and see what components have changed. Thanks!

**Comment by KevinTCoughlin** [comment]
Interesting feature add! Curious, what type of format do you need for your use case?

I ask because I wonder if we should consider accomplishing this using https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/NumberFormat.

**Comment by ethanroday** [comment]
Nothing too crazy - just percent, which would definitely be covered under `Intl.NumberFormat`.

But my thought on just making it a function was, why restrict the API? If the user wants to use number formats, they can say:

    valueFormat={val => new Intl.NumberFormat('en-US', { style: 'percent' }).format(val)}

But if they want their slider to say `15 bunnies`, they can just as easily say:

    valueFormat={val => `${val} bunnies`}

Which would be hard if it were restricted to the `Intl.NumberFormat` API.

**Comment by KevinTCoughlin** [comment]
> 
> 
> Nothing too crazy - just percent, which would definitely be covered under `Intl.NumberFormat`.
> 
> But my thought on just making it a function was, why restrict the API? If the user wants to use number formats, they can say:
> 
> ```
> valueFormat={val => new Intl.NumberFormat('en-US', { style: 'percent' }).format(val)}
> ```
> 
> But if they want their slider to say `15 bunnies`, they can just as easily say:
> 
> ```
> valueFormat={val => `${val} bunnies`}
> ```
> 
> Which would be hard if it were restricted to the `Intl.NumberFormat` API.

That's a fair point and admittedly the dev could use `Intl.NumberFormat` as part of their transform function for this prop.

File: common/changes/office-ui-fabric-react/slider-number-format_2019-01-04-19-23.json (hunk 1 above)
Comment (ethanroday) [reply common/changes/office-ui-fabric-react/slider-number-format_2019-01-04-19-23.json]: Sure thing! Done.

**Comment by ethanroday** [comment]
Thanks for the feedback, @KevinTCoughlin and @Vitalius1! Let me know if there's anything else I need to do before this can be merged.

**Comment by Vitalius1** [comment]
@ethanroday I am OK with this new feature. Thanks for the test again and for examples added. I would recommend waiting up till the end of the day so that anyone else on the team could review and then merge it.
@atneik FYI

**Comment by ethanroday** [comment]
Thanks @atneik. I do not have write access so if someone could merge that would be great!

**Comment by atneik** [comment]
I'll just wait for more eyes by EOD, and will be happy to merge it in after.

**Comment by dzearing** [comment]
Minor thought, what do you think?

```
value: number | (value: number) => string;
```

...in `render`:

```tsx
const labelText = (typeof value === 'function') ? value(value) : value;
```

This means the public contract doesn't change; it just becomes more robust. Are there downsides in just doing this?

In the slots work that @JasonGore is working on, slots can be functional like this; I wonder if textual scenarios would also benefit from this pattern.

**Comment by ethanroday** [comment]
Wouldn’t that break the slider itself? Doesn’t it need to have a numeric value?

**Comment by Vitalius1** [comment]
@dzearing in 

> `const labelText = (typeof value === 'function') ? value(value) : value;`

you are calling the `value` if it's a function and you pass `value` which is the same function? Can't seem to follow your suggestion. It looks like in case you pass a function you would have to have logic and state for the value manipulation in the consuming app doing what instead it's done in slider today. Plus to that, there is some `Math.min` used for comparing `min` with `value`.

**Comment by kenotron** [comment]
testing out the pr deploy

File: packages/office-ui-fabric-react/src/components/Slider/examples/Slider.Basic.Example.tsx [hunk 2]
```diff
@@ -39,6 +39,13 @@ export class SliderBasicExample extends React.Component<
             onChange={(value: any) => this.setState({ value })}
             showValue={true}
           />
+          <Slider
```
Comment (cliffkoh) [inline packages/office-ui-fabric-react/src/components/Slider/examples/Slider.Basic.Example.tsx]: Thanks for adding an example too :)

**Pull request merged**